### PR TITLE
[Base Parser] added post-fix conditional parser to base parser

### LINKED ISSUE #31: Base Parser can't handle ANDs & ORs

### LINKED ISSUE #74: Need unit tests for BaseParser IN support (code added in PR #73)
New support was added for quoted string conditionals as well as upper case IN.  This code has no test coverage.  Please add.

### LINKED ISSUE #80: 10.7 breaks certain "in" conditionals
This conditional is broken with latest release:
!if ("GCC49" in $(TOOL_CHAIN_TAG)) OR ("GCC5" in $(TOOL_CHAIN_TAG))
...
!endif

### PR DESCRIPTION
**Pull request by matthewfcarlson**
This allows us to handle OR, AND, and a few other cases more properly.

Fixes #80 
Fixes #31 
Fixes #74

### CODE REVIEW

File: edk2toollib/uefi/edk2/parsers/base_parser.py [hunk 1]
```diff
@@ -133,8 +133,21 @@ def ComputeResult(self, value, cond, value2):
...
         ivalue = value
         ivalue2 = value2
+        if isinstance(value, str):
+            ivalue = value.strip("\"")
+        if isinstance(value2, str):
+            ivalue2 = value2.strip("\"")
         # convert it to interpretted value
 
+        if (cond.upper() == "OR"):
+            return value or value2
```
Comment (corthon) [inline edk2toollib/uefi/edk2/parsers/base_parser.py]: Are these supposed to be value or ivalue?

File: edk2toollib/uefi/edk2/parsers/base_parser_test.py [hunk 2]
```diff
@@ -385,6 +388,27 @@ def test_process_conditional_reset(self):
         self.assertTrue(parser.InActiveCode())
         self.assertEqual(len(parser.ConditionalStack), 0)
 
+    def test_process_in_conditional(self):
+        parser = BaseParser("")
+        parser.SetInputVars({"TOOL_CHAIN_TAG": "GCC5_TEST"})
+        self.assertTrue(parser.ProcessConditional(
+            '!if ("GCC49" in $(TOOL_CHAIN_TAG)) OR ("GCC5" in $(TOOL_CHAIN_TAG))'))
+        self.assertTrue(parser.InActiveCode())
+        parser.ResetParserState()
```
Comment (corthon) [inline edk2toollib/uefi/edk2/parsers/base_parser_test.py]: Are there any new tests that exploit the added functionality? Tests that would have failed previously but now pass?

Comment (corthon) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: Did not mean to approve.

File: edk2toollib/uefi/edk2/parsers/base_parser.py (hunk 1 above)
Comment (corthon) [reply edk2toollib/uefi/edk2/parsers/base_parser.py]: Did not mean to approve.

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser.py]: Hmmm. Maybe. ivalue is meant to be the integer representation of the value. So maybe. We need to convert to a number.

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser.py]: But then we get to weird situations. What does !if 1 and TRUE mean?

File: edk2toollib/uefi/edk2/parsers/base_parser_test.py (hunk 2 above)
Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: you can see that this is a new test that would have failed on the old parser but it now works. However, we really don't have that many tests that really flex the muscles of the new conditional parser.

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: There's an issue that tracks this: #74

File: edk2toollib/uefi/edk2/parsers/base_parser.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser.py]: Alright- I decided that we should use ivalue in this situation

File: edk2toollib/uefi/edk2/parsers/base_parser_test.py (hunk 2 above)
Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: I added a few more test but more are needed. Our test coverage for base parser is pretty great (95% last time I checked) but we need more of the failure cases.

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/base_parser_test.py]: Okay further tweaks and added a few more tests

**Pull request merged**